src/util.rs

```diff
@@ -121,6 +123,61 @@ pub async fn hash_all(files: &[PathBuf], pool: &tokio::runtime::Handle) -> Resul
     Ok(hashes)
 }
 
+/// Calculate the digest of each Unix archive file in `files` on background threads in
+/// `pool`.
+///
+/// The hash is calculated by adding the filename of each archive entry followed
+/// by its contents, ignoring headers and other file metadata. This primarily
+/// exists because Apple's `ar` tool inserts timestamps for each file with
+/// no way to disable this behavior.
+pub async fn hash_all_archives(
+    files: &[PathBuf],
+    pool: &tokio::runtime::Handle,
+) -> Result<Vec<String>> {
+    let start = time::Instant::now();
+    let count = files.len();
+    let iter = files.iter().map(|path| {
+        let path = path.clone();
+        pool.spawn_blocking(move || -> Result<String> {
+            let mut m = Digest::new();
+            let reader = File::open(&path)
+                .with_context(|| format!("Failed to open file for hashing: {:?}", path))?;
+            let mut archive = Archive::new(reader);
```

Comment:
Oh, TIL that static libraries all use the `COFF` format for all three of Windows, macOS and Linux. Neat :+1: